mppe/crawler/crawler.go

```diff
@@ -0,0 +1,78 @@
+package crawler
+
+import (
+	"fmt"
+	"sync"
+)
+
+// Crawler defines how the crawlers should work
+type crawler interface {
```

Comment:
Estou com dúvidas se essa interface é mesmo necessária, principalmente que estamos falando de código idiomático Go, que preza bastante por simplicidade (mais informações sobre interfaces em Go [aqui](https://github.com/golang/go/wiki/CodeReviewComments#interfaces) e [aqui](https://golang.org/doc/effective_go.html#interfaces)). Esse jeitão de programar me lembrou bastante o design de código Java e C++.

Olhei os métodos `getURLForYear`, `getFileName` eu notei que o que muda são as categorias, você pode simplesmente criar os mapas de categorias que precisar (devidamente documentados) e ter um método que recebe o mapa e faz o que tem que fazer.

Dá uma olhada se com os outros a mesma acontece. Se for esse o caso, minha sugestão é remover a interface e simplificar bastante o código (talvez reduzir a um único arquivo): cria um mapa de mapas (sendo a chave o nome e o valor o mapa de categorias), itera nesse mapa e realizar as operações passando os mapas como parâmetro :)

Reply:
Essa sacada do mapa de mapas que eu não tava tendo, ai resolvi fazer esse strategy, valeu mesmo essa tua ideia!